staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -511,11 +511,23 @@ func (r Request) finalURLTemplate() url.URL {
 	}
 	r.params = newParams
 	url := r.URL()
-	segments := strings.Split(r.URL().Path, "/")
+	segments := strings.Split(url.Path, "/")
 	groupIndex := 0
 	index := 0
-	if r.URL() != nil && r.c.base != nil && strings.Contains(r.URL().Path, r.c.base.Path) {
-		groupIndex += len(strings.Split(r.c.base.Path, "/"))
+	if url != nil && r.c.base != nil && strings.Contains(url.Path, r.c.base.Path) {
```

Comment:
Instead of doing this, can we just do a `strings.TrimPrefix(url.Path, r.c.base.Path)`, and then try to find the "group" from the remaining string?